docs/docker-optimization/Dockerfile.optimized

```
@@ -0,0 +1,171 @@
+# Base stage for shared dependencies
+# This stage installs all dependencies for the workspace
+# and only changes when package.json or pnpm-lock.yaml changes
+FROM node:20-alpine AS base
+RUN apk add --no-cache python3 make g++
```

Hadolint (2.14.0)

[warning] Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`

(DL3018)